### PR TITLE
Feat: Auth

### PR DESCRIPTION
**Pull request by dikshadhote**
Following functionality has been added in PR using redux toolkit

1.SignUp
2. Login
3. Logout 
4. Private and public routes
5.
image
image

### CODE REVIEW

File: src/components/Login/Login.js [hunk 1]
```diff
@@ -1,5 +1,94 @@
-import React from "react";
-
+import React, { useState } from "react";
+import { Link, useNavigate } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { logIn } from "../../redux/reducers/auth/authSlice";
 export default function Login() {
-  return <div>Login</div>;
+  const [login, setLogin] = useState({
```
Comment (JantuDeb) [inline src/components/Login/Login.js:6]: You can use a better name here for states.

File: src/components/Login/Login.js [hunk 2]
```diff
@@ -1,5 +1,94 @@
...
+                className="input input-login black-light-shade-bg white-text-color"
+                id="input-password"
+                placeholder="*******"
+                type="password"
+                value="adarshBalika123"
+                onChange={(e) => {
+                  let timer = setTimeout(() => {
+                    setLogin({ ...login, password: e.target.value });
+                    clearTimeout(timer);
+                  }, 1000);
```
Comment (JantuDeb) [inline src/components/Login/Login.js:68]: Any specific reason to use timer here

**Review by JantuDeb — APPROVED** [review]
Overall  the code looks fine

**Review by SaurabhChawda — APPROVED** [review]
Design and code Looks good to me,You Can merge It.

File: src/components/Login/Login.js (hunk 2 above)
Comment (dikshadhote) [reply src/components/Login/Login.js:68]: on every click the state updates.so, to delay that

**Pull request merged**